RecoLocalFastTime/FTLClusterizer/src/MTDCPEBase.cc

```diff
@@ -0,0 +1,120 @@
+
+#include "RecoLocalFastTime/FTLClusterizer/interface/MTDCPEBase.h"
+
+// MessageLogger
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+// Magnetic field
+#include "MagneticField/Engine/interface/MagneticField.h"
+
+
+#include <iostream>
+
+using namespace std;
+
+//-----------------------------------------------------------------------------
+//  A constructor run for generic and templates
+//
+//-----------------------------------------------------------------------------
+MTDCPEBase::MTDCPEBase(edm::ParameterSet const & conf,
+		       const MTDGeometry& geom)
+  : geom_(geom)
+{
+   fillDetParams();
+}
+
+
+//-----------------------------------------------------------------------------
+//  Fill all variables which are constant for an event (geometry)
+//-----------------------------------------------------------------------------
+void MTDCPEBase::fillDetParams()
+{
+  auto const & dus = geom_.detUnits();
+  unsigned detectors = dus.size();
+  m_DetParams.resize(detectors);
+  LogDebug("MTDCPEBase::fillDetParams():") <<"caching "<<detectors<<"MTD detectors"<<endl;
+  for (unsigned i=0; i!=detectors;++i) 
+    {
+      auto & p=m_DetParams[i];
+      p.theDet = dynamic_cast<const MTDGeomDetUnit*>(dus[i]);
+      assert(p.theDet);
+      
+      p.theOrigin = p.theDet->surface().toLocal(GlobalPoint(0,0,0));
+      
+      //--- p.theDet->type() returns a GeomDetType, which implements subDetector()
+      p.thePart = p.theDet->type().subDetector();
+            
+      //--- bounds() is implemented in BoundSurface itself.
+      p.theThickness = p.theDet->surface().bounds().thickness();
+      
+      // Cache the det id for templates and generic erros
+      p.theTopol = &(static_cast<const ProxyMTDTopology&>(p.theDet->topology()));
+      assert(p.theTopol);
+      p.theRecTopol = &(static_cast<const RectangularMTDTopology&>(p.theTopol->specificTopology()));    
+      assert(p.theRecTopol);
+      
+      //--- The geometrical description of one module/plaquette
+      std::pair<float,float> pitchxy = p.theRecTopol->pitch();
+      p.thePitchX = pitchxy.first;	     // pitch along x
+      p.thePitchY = pitchxy.second;	     // pitch along y
+            
+      LogDebug("MTDCPEBase::fillDetParams()") << "***** MTD LAYOUT *****"
+					      << " thePart = " << p.thePart
+					      << " theThickness = " << p.theThickness
+					      << " thePitchX  = " << p.thePitchX
+					      << " thePitchY  = " << p.thePitchY;
+    }
+}
+
+//-----------------------------------------------------------------------------
+//  One function to cache the variables common for one DetUnit.
+//-----------------------------------------------------------------------------
+void
+MTDCPEBase::setTheClu( DetParam const & dp, ClusterParam & cp ) const
+{   
+}
+
+std::unique_ptr<MTDCPEBase::ClusterParam> MTDCPEBase::createClusterParam(const FTLCluster & cl) const
+{
+  return std::make_unique<ClusterParam>(cl);
```

Comment:
static analyzer flags this as a memory leak 
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25063/32286/llvm-analysis/report-b62abd.html#EndPath

Reply:
The static analyzer is wrong.